rpc/wrtc_server_stream.go

```diff
@@ -277,13 +277,19 @@ func (s *webrtcServerStream) processHeaders(headers *webrtcpb.RequestHeaders) {
 		}
 	}
 
+	s.ch.server.processHeadersMu.RLock()
+	s.ch.server.processHeadersWorkers.Add(1)
+	s.ch.server.processHeadersMu.RUnlock()
+
+	// Check if context has errored: underlying server may have been `Stop`ped,
+	// in which case we mark this processHeaders worker as `Done` and return.
+	if err := s.ch.server.ctx.Err(); err != nil {
+		s.ch.server.processHeadersWorkers.Done()
+		return
+	}
+
 	// take a ticket
 	select {
-	case <-s.ch.server.ctx.Done():
```

Comment:
This `case` has been replaced by the `if` check above. I'm no longer `closeWithSendError`ing, as `Stop` should be closing all underlying peer connections. I can't quite remember why/if we thought this was a good/acceptable change, @dgottlieb; let me know if what I have seems incorrect.

Reply:
Looks good to me